stdlib/public/core/UnsafeRawBufferPointer.swift.gyb

```diff
@@ -348,6 +348,56 @@ public struct Unsafe${Mutable}RawBufferPointer
     return 0
   }
 
+  %  if mutable:
+    /// Initializes memory in the buffer with the elements of
+    /// `source` and binds the initialized memory to type `T`.
+    ///
+    /// Returns an iterator to any elements of `source` that didn't fit in the 
+    /// buffer, and an index to the point in the buffer one past the last element
+    /// written.
+    ///
+    /// - Precondition: The memory in `startIndex..<(source.count *
+    ///   MemoryLayout<T>.stride)` is uninitialized or initialized to a
+    ///   trivial type. The buffer must contain sufficient memory to
+    ///   accommodate at least `source.underestimateCount` elements.
+    ///
+    /// - Postcondition: The memory at `self[startIndex..<returned index] *
+    ///   MemoryLayout<T>.stride` is bound to type `T`.
+    ///
+    /// - Postcondition: The `T` values at `self..<self + source.count *
+    ///   MemoryLayout<T>.stride` are initialized.
+    ///
+    // TODO: Optimize where `C` is a `ContiguousArrayBuffer`.
+    public func initializeMemory<S: Sequence>(
+      as: S.Iterator.Element.Type, from source: S
+    ) -> (S.Iterator, Index) {
+      let underestimate = source.underestimatedCount
+      guard let base = self.baseAddress else {
+        _precondition(underestimate == 0, "no memory available to initialize from source")
+        return (source.makeIterator(),startIndex)
+      }
+      _precondition(underestimate <= count / MemoryLayout<S.Iterator.Element>.stride,
+        "insufficient space to accommodate source.underestimatedCount elements")
+    
+      var it = source.makeIterator()
+      var idx = self.startIndex
+
+      for p in stride(from: base, 
+        // only advance to as far as the last element that will fit
+        to: base + count - (count % MemoryLayout<S.Iterator.Element>.stride), 
```

Comment:
Isn't this equivalent to `base + count`?  This is basically rounding count down to the nearest multiple of stride, but since we're incrementing by stride we'll always stop at a multiple of stride. It doesn't randomly include `to` if that's not a multiple.